components/markdown/src/codeblock/mod.rs

```diff
@@ -112,19 +119,28 @@ impl<'config> CodeBlock<'config> {
             highlighter.pre_style(),
             highlighter.pre_class(),
             fence.line_numbers,
+            fence.enable_copy,
         );
         Ok((
             Self {
                 highlighter,
                 line_numbers: fence.line_numbers,
                 line_number_start: fence.line_number_start,
-                highlight_lines: fence.highlight_lines,
-                hide_lines: fence.hide_lines,
+                highlight_lines: fence.highlight_lines.clone(),
+                hide_lines: fence.hide_lines.clone(),
+                include: fence.include.map(|s| s.to_string()),
             },
             html_start,
         ))
     }
 
+    pub fn include(&self, base: Option<&PathBuf>) -> Option<String> {
+        let path = base?.join(self.include.as_ref()?);
+        let res = read_file(&path);
+
+        res.ok()
```

Comment:
It may be better to add a warning when the path doesn't exist, instead of silently failing?
Would help debugging missing paths.